### PR TITLE
Add pipeline.proto file

### PR DESCRIPTION
**Pull request by difince**
Add just the v2beta1 pipeline.proto file 

No auto-generated file for easy review.

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

**Description of your changes:**

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/api/v2beta1/pipeline.proto
Comment (annajung) [inline backend/api/v2beta1/pipeline.proto]: line 108 needs to be updated too? 
```
- post: "/apis/v2beta1/pipeline_versions"
+ post: "/apis/v2beta1/pipelines/{pipeline_id}/pipeline_versions"
```

Comment (annajung) [reply backend/api/v2beta1/pipeline.proto]: Also I believe it was `/versions` not `/pipeline_versions` since pipeline part is inferred

File: backend/api/v2beta1/pipeline.proto
Comment (annajung) [inline backend/api/v2beta1/pipeline.proto]: nit
```suggestion
  //TODO v2: Try to fix this! See https://github.com/kubeflow/pipelines/issues/7982
```

File: backend/api/v2beta1/pipeline.proto
Comment (annajung) [inline backend/api/v2beta1/pipeline.proto]:
```suggestion
      get: "/apis/v2beta1/pipelines/names/{name}"
```

File: backend/api/v2beta1/pipeline.proto
Comment (difince) [reply backend/api/v2beta1/pipeline.proto]: Agree , Thank you, @annajung

**Comment by annajung** [comment]
lgtm! thank you!

File: backend/api/v2beta1/pipeline.proto
Comment (Linchin) [inline backend/api/v2beta1/pipeline.proto]: Make the comment format consistent.

File: backend/api/v2beta1/pipeline.proto
Comment (Linchin) [inline backend/api/v2beta1/pipeline.proto]: Order the field numbers.

File: backend/api/v2beta1/pipeline.proto
Comment (Linchin) [inline backend/api/v2beta1/pipeline.proto:125]: nit: maybe move the `DeletePipelineVersion` to the last rpc definition, to follow CRUD order?

**Review by Linchin — APPROVED** [review]

**Comment by annajung** [comment]
Before merging, see https://github.com/difince/pipelines/pull/6/files#r946169670

**Comment by difince** [comment]
> Before merging, see https://github.com/difince/pipelines/pull/6/files#r946169670

Why .. I see only the removal of resource_reference.proto as a possible problem?

**Comment by annajung** [comment]
Sorry, I don't know why that link is not working -- but it's pointing to the discussion about parameter.proto which is used in the pipeline.proto.

**Comment by Linchin** [comment]
for `repeated Parameter parameters = 5;
`, maybe use ` map<string, google.protobuf.Value> parameters = 5;` instead? Thus we don't have to use the `parameter.proto`, and we have the same format for all parameter fields.

**Comment by difince** [comment]
> for repeated Parameter parameters = 5; , maybe use `map<string, google.protobuf.Value> parameters = 5`; instead? Thus we don't have to use the `parameter.proto`, and we have the same format for all parameter fields.

I do not have a strong opinion about this. It looks like it will work but hope we are not going to confuse the clients as the `google.protobuf.Values`  could be one of 

```
message Value {
  // The kind of value.
  oneof kind {
    // Represents a null value.
    NullValue null_value = 1;
    // Represents a double value.
    double number_value = 2;
    // Represents a string value.
    string string_value = 3;
    // Represents a boolean value.
    bool bool_value = 4;
    // Represents a structured value.
    Struct struct_value = 5;
    // Represents a repeated `Value`.
    ListValue list_value = 6;
  }
}
```
Anyway, I will make `pipeline.proto` file using `map<string, google.protobuf.Value>` .

**Review by annajung — APPROVED** [review]
lgtm after the removing parameter.proto

**Comment by difince** [comment]
closing this PR as not relevant anymore

**Pull request closed without merging**
